Infrastructure/TuitionPartnerService.cs

```diff
@@ -39,15 +115,22 @@ public IEnumerable<TuitionPartnerResult> OrderTuitionPartners(IEnumerable<Tuitio
 
             case TuitionPartnerOrderBy.MinPrice:
                 return ordering.Direction == OrderByDirection.Descending
-                    ? results.OrderBy(e => e.Name).OrderByDescending(e => e.Prices.Min(x => x.HourlyRate))
-                    : results.OrderBy(e => e.Name).OrderBy(e => e.Prices.Min(x => x.HourlyRate));
+                    ? results
+                        .OrderBy(e => e.Name)
+                        .OrderBy(s => (ordering.SeoUrlOrderBy == null || ordering.SeoUrlOrderBy.Length == 0) ? -1 : Array.IndexOf(ordering.SeoUrlOrderBy, s.SeoUrl))
```

Comment:
Using multiple sequential `OrderBy` invocations in a query will overwrite the previous sort order, and only the last `OrderBy` will take effect. To sort by multiple criteria, you should use `OrderBy` followed by one or more `ThenBy` clauses. `ThenBy` is used to sort by additional criteria after the initial sort order has been established with `OrderBy`.

Reply:
Done for all the sorts, although this changes the "repeatable random" ordering that was used.